src/tools/cephfs/cephfs-shell

```diff
@@ -779,6 +779,148 @@ class CephFSShell(Cmd):
 
         return 0
 
+    def complete_ln(self, text, line, begidx, endidx):
+        """
+        auto complete of file name.
+        """
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    ln_parser = argparse.ArgumentParser(
+        description='Create a hard link or a symbolic '
+                    'link to an existing file or directory.')
+    ln_parser.add_argument('src_path', type=str, action=path_to_bytes,
+                           help='Path of the source file')
+    ln_parser.add_argument('dest_path', type=str, action=path_to_bytes,
+                           help='Destination path', nargs='?')
+    ln_parser.add_argument('-s', '--symbolic', action='store_true',
+                           help='Create symbolic link')
+    ln_parser.add_argument('-v', '--verbose', action='store_true',
+                           help='Print name of each linked file')
+    ln_parser.add_argument('-f', '--force', action='store_true',
+                           help='Force create link/symbolic link')
+
+    @with_argparser(ln_parser)
+    def do_ln(self, args):
+        # check if source file/dir exists.
+        if not is_file_exists(args.source_path) \
+                and not is_dir_exists(args.source_path):
+            set_exit_code_msg(errno.ENOENT,
+                              msg=f"ln: failed to access "
+                                  f"'{args.source_path.decode('utf-8')}"
+                                  f"': No such file or directory")
+            return
+        source_path = args.source_path
+
+        destination_path = ''
+        # create link at CWD
+        if args.destination_path is None:
+            cwd = cephfs.getcwd().decode('utf-8')
+            # sometimes, cwd is like /dir1/dir2, and there is a source file
+            # foo then path would be like /dir1/dir2foo which is not expected
+            # therefore checking existence of `/` here.
+            if cwd[len(cwd) - 1] != '/':
+                cwd = cwd + '/'
+            cwd = cwd.encode('utf-8')
+            # regex that would fetch the last part of a path
```

Comment:
Can we use `os.path.basename()` here? and other places where you are finding basepath from full path

Reply:
Done